common/sphinx.c

```diff
@@ -78,6 +87,18 @@ struct sphinx_path *sphinx_path_new_with_key(const tal_t *ctx,
 	return sp;
 }
 
+bool sphinx_path_set_rendezvous(struct sphinx_path *sp,
+				const struct node_id *rendezvous_id)
+{
+	if (rendezvous_id == NULL) {
+		sp->rendezvous_id = tal_free(sp->rendezvous_id);
+		return true;
+	} else {
+		sp->rendezvous_id = tal(sp, struct pubkey);
```

Comment:
Potential leak here if set twice?

I think start with sp->rendezvous_id = tal_free(...) before checking if the new value is NULL?